Xamarin.Forms.Core/Application.cs

```diff
@@ -36,6 +36,20 @@ public Application()
 			_platformConfigurationRegistry = new Lazy<PlatformConfigurationRegistry<Application>>(() => new PlatformConfigurationRegistry<Application>(this));
 		}
 
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public override IEnumerable<T> WalkChildren<T>()
+		{
+			var pages = NavigationProxy?.ModalStack;
+			if (pages == null)
+				return base.WalkChildren<T>();
+
+			var result = new List<T>(base.WalkChildren<T>());
+			foreach (var page in pages)
```

Comment:
Will this check all of the pages on the stack? It seems like we should only be worried about the tab indices for items on the top page of the stack.

Reply:
I think so. It will be enough to know the page on which the element is located.